python/lambdas/copy-files-from-tdr/lambda_function.py

```diff
@@ -1,23 +1,79 @@
 import os
 import json
+import uuid
+from datetime import datetime
+
 import boto3
+import botocore
+import jsonschema
+from botocore.exceptions import ClientError
+from jsonschema import validate
+from tdr_json_schema import incoming_schema
 
 s3_client = boto3.client('s3')
 sqs_client = boto3.client('sqs')
 
-
 def lambda_handler(event, context):
     destination_bucket = os.environ["DESTINATION_BUCKET"]
     destination_queue = os.environ["DESTINATION_QUEUE"]
     for record in event['Records']:
         body = json.loads(record['body'])
         file_id = body['fileId']
         source_bucket = body['bucket']
+        assert_objects_exist(source_bucket, file_id)
+        validate_metadata(source_bucket, f"{file_id}.metadata")
```

Comment:
@MancunianSam I don't like that we're inferring a metadata file exists from the message. Can we add tech debt to us and TDR to fix this by including both object refs in the message.

Reply:
\+ if we are inferring, let's do it once and use it as a variable.

Reply:
Yeah fair point, I'll do it soon-ish. I've reached my TDR fiddling quota for the month.